beacon_node/execution_layer/src/lib.rs

```diff
@@ -1183,15 +1200,17 @@ impl<T: EthSpec> ExecutionLayer<T> {
 
                 if local.should_override_builder().unwrap_or(false) {
                     let percentage_difference =
-                        percentage_difference_u256(local_value, relay_value);
+                        percentage_difference_u256(local_value, boosted_relay_value);
```

Comment:
Same with the `ignore_builder_override_suggestion_threshold` config, this change would alter the calculation result for this existing feature.

I wonder if we should deprecate these flags in the BN, as these overrides break compliance with the spec, because we are ignoring the `builder_comparison_factor` to some extent when these flags are present:
> This parameter is only relevant if the beacon node is connected to a builder, deems it safe to produce a builder payload, and receives valid responses from both the builder endpoint _and_ the paired execution node. When these preconditions are met, the server MUST act as follows:
>  * if `exec_node_payload_value >= builder_comparison_factor * (builder_payload_value // 100)`, then return a full (unblinded) block containing the execution node payload.
>  * otherwise, return a blinded block containing the builder payload header.

and it might make more sense to have any builder related configurations in the VC, and have the VC provide a `builder_boost_factor` derived from these configurations.

Reply:
Correct me if I'm wrong - I think the `local_value >= boosted_relay_value` should be the only comparison in the BN in order to be fully compliant with the spec.

Reply:
The spec doesn't prohibit other BN-level flags influencing the behaviour, so long as those flags are reasonable. I was kind of leaning towards ditching `always-prefer-builder-payload` and `builder-profit-threshold`, but keeping `ignore-builder-override-suggestion-threshold` for now.